styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/PublisherHandler.java

```diff
@@ -40,15 +41,14 @@
 import com.spotify.styx.state.RunState;
 import java.util.Objects;
 import java.util.concurrent.Callable;
-import java.util.function.BiConsumer;
 import javaslang.control.Try;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
  * An event consumer that integrates {@link RunState.State} values with a {@link Publisher}.
  */
-public class PublisherHandler implements BiConsumer<SequenceEvent, RunState> {
+public class PublisherHandler implements StyxScheduler.EventConsumer {
```

Comment:
This however creates a circular dependency. How about we make `StyxScheduler.EventConsumer` a proper interface?

Reply:
👍